river/stream/twitch_chat_stream.py

```diff
@@ -0,0 +1,167 @@
+            A list of channel names like ["asmongold", "shroud"] you want to collect messages from.
+        buffer_size
+            Size of buffer in kB used for receiving responses from Twitch with IRC (default 2 kB).
+        timeout
+            A timeout value in seconds for waiting response from Twitch (default 5s). It can be usefull if all channels requested are offline.
+        dt_format
+            A format of datetime string in stream item as used by datetime.strftime (look at https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes
+            for format codes if needed). Default format set to ISO 8601 like YYYY-MM-DDThh:mm:ss.
+
+    Returns
+    -------
+    Stream items are python dictionaries with keys:
+    dt (str)
+        Datetime string indicating when response with this message has been received from Twitch.
+    channel (str)
+        A channel where message has been posted.
+    username (str)
+        User who posted the message.
+    msg (str)
+        The message itself.
+
+    Examples
+    --------
+    The live stream is instantiated by passing your Twitch account nickname, OAuth token and list of channels. Other parameters are optional.
+
+    >>> from river import stream
+
+    >>> twitch_chat = stream.TwitchChatStream(
+    ...     nickname="twitch_user1",
+    ...     token="oauth:okrip6j6fjio8n5xpy2oum1lph4fbve",
+    ...     channels=["asmongold", "shroud"]
+    ... )
+
+    The stream can be iterated over like this:
+
+    ```py
+    for item in twitch_chat:
+        print(item)
+    ```
+
+    Here's a single stream item example:
+    ```py
+    {
+        'dt': '2022-09-14T10:33:37',
+        'channel': 'pgl_dota2',
+        'username': 'moojiejaa',
+        'msg': 'damn this chat mod are wild'
+    }
+    ```
+
+    References
+    ----------
+    [^1]: [Twitch IRC doc](https://dev.twitch.tv/docs/irc)
+    """
+
+    def __init__(
+        self,
+        nickname: str,
+        token: str,
+        channels: List[str],
+        buffer_size: int = BUFFER_SIZE,
+        timeout: int = TIMEOUT,
+        dt_format: str = DT_FORMAT,
+    ):
+        self.nickname = nickname
+        self.token = token
+        self.channels = channels
+        self.buffer_size = buffer_size
+        self.timeout = timeout
+        self.dt_format = dt_format
+
+    def _send(self, s: socket.socket, msg: IrcMessage, payload: Optional[str] = None) -> None:
+        text = msg.name
+        if payload:
+            text += f" {payload}"
+        s.send(f"{text}\n".encode(ENCODING))
+
+    def _setup_connection(self, s: socket.socket) -> None:
+        s.connect((SERVER, PORT))
+        self._send(s, IrcMessage.PASS, self.token)
+        self._send(s, IrcMessage.NICK, self.nickname)
+        for chan in self.channels:
+            self._send(s, IrcMessage.JOIN, f"#{chan}")
+        s.settimeout(self.timeout)
+
+    def _is_ping(self, resp: str) -> bool:
+        return bool(re.search(PING_PATTERN, resp))
```

Comment:
I believe you can compile the regex pattern and that usually it's faster that way. Like:

```py
ping_pattern = re.compile(pattern)
ping_pattern.search(resp)
```